x/evm/wrappers/bank.go

```diff
@@ -0,0 +1,110 @@
+
+// MintAmountToAccount converts the given amount into the evm coin scaling
+// the amount to the original decimals, then mints that amount to the provided account.
+func (w BankWrapper) MintAmountToAccount(ctx context.Context, recipientAddr sdk.AccAddress, amt math.Int) error {
+	coin := sdk.NewCoin(config.GetEVMCoinDenom(), amt)
+
+	convertedCoin, err := convertEvmCoinFrom18Decimals(coin)
+	if err != nil {
+		return errors.Wrap(err, "failed to mint coin to account in bank wrapper")
+	}
+
+	coinsToMint := sdk.NewCoins(convertedCoin)
+	if err := w.MintCoins(ctx, types.ModuleName, coinsToMint); err != nil {
+		return errors.Wrap(err, "failed to mint coins to account in bank wrapper")
+	}
+
+	return w.SendCoinsFromModuleToAccount(ctx, types.ModuleName, recipientAddr, coinsToMint)
+}
+
+// BurnAmountFromAccount convert the given amount into the evm coin scaling
+// the amount to the original decimals, then burns that quantity from the provided account.
+func (w BankWrapper) BurnAmountFromAccount(ctx context.Context, account sdk.AccAddress, amt math.Int) error {
+	coin := sdk.NewCoin(config.GetEVMCoinDenom(), amt)
+
+	convertedCoin, err := convertEvmCoinFrom18Decimals(coin)
+	if err != nil {
+		return errors.Wrap(err, "failed to burn coins from account in bank wrapper")
+	}
+
+	coinsToBurn := sdk.NewCoins(convertedCoin)
+	if err := w.SendCoinsFromAccountToModule(ctx, account, types.ModuleName, coinsToBurn); err != nil {
+		return errors.Wrap(err, "failed to burn coins from account in bank wrapper")
+	}
+	return w.BurnCoins(ctx, types.ModuleName, coinsToBurn)
+}
+
+// ------------------------------------------------------------------------------------------
+// Bank keeper shadowed methods
+// ------------------------------------------------------------------------------------------
+
+// GetBalance returns the balance of the given account converted to 18 decimals.
+func (w BankWrapper) GetBalance(ctx context.Context, addr sdk.AccAddress, denom string) sdk.Coin {
+	// Get the balance from the BankModule. The balance returned is in the bank
+	// decimals representation, which could be different than the 18 decimals
+	// representation used in the evm.
+	coin := w.BankKeeper.GetBalance(ctx, addr, denom)
+
+	return mustConvertEvmCoinTo18Decimals(coin)
+}
+
+// // SendCoinsFromAccountToModule wraps around the Cosmos SDK x/bank module's
+// // SendCoinsFromAccountToModule method to convert the evm coin, if present in
+// // the input, to its original representation.
+// func (w BankWrapper) SendCoinsFromAccountToModule(ctx context.Context, senderAddr sdk.AccAddress, recipientModule string, coins sdk.Coins) error {
+// 	convertedCoins, err := convertCoinsFrom18Decimals(coins)
+// 	if err != nil {
+// 		return errors.Wrap(err, "failed to send coins to module in bank wrapper")
+// 	}
+//
+// 	return w.BankKeeper.SendCoinsFromAccountToModule(ctx, senderAddr, recipientModule, convertedCoins)
+// }
```

Comment:
we need this function for the fees deduction

Reply:
ATM it is not used, that' why I commented it out

Reply:
it is [used by the sdk ante handler](https://github.com/evmos/evmos/blob/fb3d693aca261e2035d8ed629a9bc9f975f54dde/x/evm/keeper/fees.go#L61)

Reply:
VS Code 1, Vim zero...it was not found in my LSP.